Tests/FluentAssertions.Specs/Collections/Data/DataColumnCollectionAssertionExtensionsSpecs.cs

```diff
@@ -0,0 +1,438 @@
+
+                // Act & Assert
+                columnCollection1.Should().NotBeSameAs(columnCollection2);
+            }
+        }
+
+        public class HaveSameCount
+        {
+            [Fact]
+            public void When_subject_is_null_it_should_fail()
+            {
+                // Arrange
+                var subject = default(DataColumnCollection);
+
+                var expectation = new DataTable().Columns;
+
+                // Act
+                Action action =
+                    () => subject.Should().HaveSameCount(expectation, because: "we {0}", "care");
+
+                // Assert
+                action.Should().Throw<XunitException>().WithMessage(
+                    "Expected * to have the same count as * because we care, but found <null>.*");
+            }
+
+            [Fact]
+            public void When_expectation_is_null_it_should_fail()
+            {
+                // Arrange
+                var dataTable = new DataTable();
+
+                for (int seed = 0; seed < 3; seed++)
+                {
+                    dataTable.Columns.Add(CreateTestDataColumn(seed));
+                }
+
+                var nullReference = default(DataColumnCollection);
+
+                // Act
+                Action action =
+                    () => dataTable.Columns.Should().HaveSameCount(nullReference);
+
+                // Assert
+                action.Should().Throw<ArgumentNullException>().WithMessage(
+                    "Cannot verify count against a <null> collection.*");
+            }
+
+            public class DataColumnCollectionAssertions
+            {
+                [Fact]
+                public void When_two_collections_have_the_same_number_of_columns_it_should_succeed()
+                {
+                    // Arrange
+                    var firstDataTable = new DataTable();
+                    var secondDataTable = new DataTable();
+
+                    for (int seed = 0; seed < 3; seed++)
+                    {
+                        firstDataTable.Columns.Add(CreateTestDataColumn(seed));
+                        secondDataTable.Columns.Add(CreateTestDataColumn(seed + 10));
+                    }
+
+                    // Act & Assert
+                    firstDataTable.Columns.Should().HaveSameCount(secondDataTable.Columns);
+                }
+
+                [Fact]
+                public void When_two_collections_do_not_have_the_same_number_of_columns_it_should_fail()
+                {
+                    // Arrange
+                    var firstDataTable = new DataTable();
+                    var secondDataTable = new DataTable();
+
+                    for (int seed = 0; seed < 3; seed++)
+                    {
+                        firstDataTable.Columns.Add(CreateTestDataColumn(seed));
+                        secondDataTable.Columns.Add(CreateTestDataColumn(seed + 10));
+                    }
+
+                    secondDataTable.Columns.RemoveAt(1);
```

Comment:
Bonus, removing the loops also removes the need for `RemoveAt`

Reply:
Hmm, yeah, for small row counts it's not hugely different:
```
                    for (int seed = 0; seed < 3; seed++)
                    {
                        firstDataTable.Columns.Add(CreateTestDataColumn(seed));
                        secondDataTable.Columns.Add(CreateTestDataColumn(seed + 10));
                    }

                    secondDataTable.Columns.RemoveAt(1);
```
vs.
```
                    firstDataTable.Columns.Add(CreateTestDataColumn(1));
                    firstDataTable.Columns.Add(CreateTestDataColumn(2));
                    firstDataTable.Columns.Add(CreateTestDataColumn(3));

                    secondDataTable.Columns.Add(CreateTestDataColumn(11));
                    secondDataTable.Columns.Add(CreateTestDataColumn(13));
```
Though, I do think that having the `RemoveAt` call actually makes the test _more explicit_; when reading the test, it is _super obvious_ that the collections are intended to be different.